src/main/java/org/mule/module/xml/internal/xpath/XPathEvaluator.java

```diff
@@ -115,15 +118,23 @@ public XPathEvaluator(String expression, XPathFactory xpathFactory, Collection<N
     for (int i = 0; i < size; i++) {
       StringWriter sw = new StringWriter();
       toString.transform(new DOMSource(nodeList.item(i)), new StreamResult(sw));
-      strings.add(sw.toString());
+      // Since Saxon 9.9.1-1, the "indented" serializer options seems to add a new-line character
+      // after each DOM Node. Could not found a property to avoid this behaviour.
+      strings.add(cleanTrailingNewlineIfNecessary(sw.toString()));
     }
 
     return strings;
   }
 
+  private String cleanTrailingNewlineIfNecessary(String line) {
+    if (!keepTrailingNewlines && line.endsWith("\n")) {
+      line = line.substring(0, line.lastIndexOf("\n"));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Put single-quotes around '\n' to use the faster "lastIndexOf(char)" method. ([squid:S3027](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3027))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-xml-module&pullRequest=29&issues=AWilgRG3otIXnmsUpHVi&open=AWilgRG3otIXnmsUpHVi)

[ISSUE_KEY]: AWilgRG3otIXnmsUpHVi